great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4685,34 +4681,23 @@ def _scaffold_root_conf(cls) -> bool:
 
     def _init_datasources(self) -> None:
         """Initialize the datasources in store"""
+        self._datasources = {}
```

Comment:
Would like to leverage this `UserDict` impl for this too but will come back later if anything - minimizing scope to just focus on resolving agent issues